hack/ensure-azcli.sh

```diff
@@ -25,5 +25,16 @@ if [[ -z "$(command -v az)" ]]; then
   AZ_REPO=$(lsb_release -cs)
   echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ ${AZ_REPO} main" | tee /etc/apt/sources.list.d/azure-cli.list
   apt-get update && apt-get install -y azure-cli
-  az login --service-principal -u "${AZURE_CLIENT_ID}" -p "${AZURE_CLIENT_SECRET}" --tenant "${AZURE_TENANT_ID}" > /dev/null
+
+  if [[ -n "${AZURE_FEDERATED_TOKEN_FILE:-}" ]]; then
+    # AZURE_CLIENT_ID has been overloaded with Azure Workload ID in the preset-azure-cred-wi.
+    # This is done to avoid exporting Azure Workload ID as AZURE_CLIENT_ID in the test scenarios.
+    az login --service-principal -u "${AZURE_CLIENT_ID}" -t "${AZURE_TENANT_ID}" --federated-token "$(cat "${AZURE_FEDERATED_TOKEN_FILE}")" > /dev/null
+
+    # Use --auth-mode "login" in az storage commands.
+    ENABLE_AUTH_MODE_LOGIN="true"
+    export ENABLE_AUTH_MODE_LOGIN
```

Comment:
You can assign and export in one line, just fyi:
```shell
export ENABLE_AUTH_MODE_LOGIN="true"
```

Reply:
I got a diff to use in the upcoming CAPZ test migration PR :)

Reply:
Sometimes linter complains about assign + export in one line though!